src/plugins/data/public/search/aggs/agg_config.ts

```diff
@@ -27,18 +28,20 @@ import { ISearchSource } from '../search_source';
 import { FieldFormatsContentType, KBN_FIELD_TYPES } from '../../../common';
 import { FieldFormatsStart } from '../../field_formats';
 
-export interface AggConfigOptions {
-  type: IAggType;
+export interface AggConfigJson {
+  type: string;
   enabled?: boolean;
   id?: string;
-  params?: Record<string, any>;
+  params?: Record<string, any>; // TODO: this should enforce serializable state
```

Comment:
whole AggConfigOptions should be serializable ....

Reply:
AggConfigSerialized ? SerializedAggConfig ?

Reply:
my vote is `AggConfigSerialized`, will update this